src/contract_wrappers/contract_wrapper.ts

```diff
@@ -1,13 +1,46 @@
 import * as _ from 'lodash';
 import * as Web3 from 'web3';
 import {Web3Wrapper} from '../web3_wrapper';
-import {ZeroExError, Artifact} from '../types';
+import {AbiDecoder} from '../utils/abi_decoder';
+import {
+    ZeroExError,
+    Artifact,
+    LogWithDecodedArgs,
+    RawLog,
+    ContractEvents,
+    SubscriptionOpts,
+    IndexedFilterValues,
+} from '../types';
 import {utils} from '../utils/utils';
 
 export class ContractWrapper {
     protected _web3Wrapper: Web3Wrapper;
-    constructor(web3Wrapper: Web3Wrapper) {
+    private _abiDecoder?: AbiDecoder;
+    constructor(web3Wrapper: Web3Wrapper, abiDecoder?: AbiDecoder) {
         this._web3Wrapper = web3Wrapper;
+        this._abiDecoder = abiDecoder;
+    }
+    protected async _getLogsAsync(address: string, eventName: ContractEvents, subscriptionOpts: SubscriptionOpts,
+                                  indexFilterValues: IndexedFilterValues,
+                                  abi: Web3.ContractAbi): Promise<LogWithDecodedArgs[]> {
```

Comment:
Let's use a type alias for `Web3.ContractAbi`

Reply:
It's an internal protected function. We don't show it in the docs